l1-contracts/deploy-scripts/upgrade/v30/EcosystemUpgrade_v30.s.sol

```diff
@@ -53,12 +53,15 @@ contract EcosystemUpgrade_v30 is Script, DefaultEcosystemUpgrade {
     // }
 
     function registerBridgedTokensInNTV(address _bridgehub) public {
-        INativeTokenVault ntv = INativeTokenVaultBase(IBridgehubBase(_bridgehub).assetRouter().nativeTokenVault());
+        NativeTokenVaultBase ntv = NativeTokenVaultBase(
+            address(IL1AssetRouter(address(IBridgehubBase(_bridgehub).assetRouter())).nativeTokenVault())
+        );
         address[] memory savedBridgedTokens;
         /// todo get save bridged tokens.
         /// for tokens in the bridged token list
         for (uint256 i = 0; i < savedBridgedTokens.length; ++i) {
-            address token = ntv.bridgedTokens(i);
+            // TODO it's cludge to convert from bytes32 to address, need to have proper solution
+            address token = address(uint160(uint256(ntv.bridgedTokens(i))));
```

Comment:
its bytes32 [assetId](https://github.com/matter-labs/era-contracts/blob/5b2e8f22711d372640ad629e4d6eee525d56bcfc/l1-contracts/contracts/bridge/ntv/NativeTokenVaultBase.sol#L69). Probably this file is never compiled properly. Please undo this

Reply:
I think the best solution is to add running the upgradeTestV30 to CI in a foundry test, please do that

Reply:
yes, the problem is `function addLegacyTokenToBridgedTokensList(address _token)` and it's compilable. I assume the test-v30 is part of the tests and it's just compiled. but doing nothing else

Reply:
discussed the switching the test to V30 makes it compile